### ISSUE TITLE
DataError running migration wagtailcore.0048_add_default_workflows

### ISSUE THREAD

**Issue by maribedran** [wagtail/wagtail#10895]
### Issue Summary

The 0048 migration tries to create a `GroupApprovalTask` with a name that joins the names of all existing groups with publish permissions. Running the migration on an existing project with slightly long group names causes the migration to fail.

In order to run the migration, I had to update all the group names and then update them again to the original values. Ideally, the migration code should truncate the name to respect the field's max length.

`wagtail/migrations/0048_add_default_workflows.py`
```
...
            group_names = " ".join([group.name for group in groups])
            task = GroupApprovalTask.objects.create(
                name=group_names + " approval",
                content_type=group_approval_content_type,
                active=True,
            )
            task.groups.set(groups)
```

```
Applying wagtailcore.0048_add_default_workflows...Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(255)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/wagtail_modeltranslation/management/commands/migrate_translation.py", line 27, in handle
    super(Command, self).handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/usr/local/lib/python3.9/site-packages/wagtail/migrations/0048_add_default_workflows.py", line 58, in create_default_workflows
    task = GroupApprovalTask.objects.create(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 453, in create
    obj.save(force_insert=True, using=self.db)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 739, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 775, in save_base
    parent_inserted = self._save_parents(cls, using, update_fields)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 804, in _save_parents
    updated = self._save_table(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 881, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 919, in _do_insert
    return manager._insert(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 1270, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1416, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.DataError: value too long for type character varying(255)
```

### Steps to Reproduce

1. Create enough auth groups with publish permission so that their names joined exceed 255 characters
2. Upgrade to Wagtail 5 and run migrations

Any other relevant information. For example, why do you consider this a bug, and what did you expect to happen instead?

The migration code uses existing data from the database but doesn't respect the field's constraint when creating a new object.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: No, the issue necessarily happens on a project with existing data.

### Technical details

-   Python version: 3.9
-   Django version: 3.2.21 
-   Wagtail version: wagtail==5.1.1
-   Browser version: Not applicable

**Comment by maribedran** [comment]
If this helps anyone who encountered the same problem, here's the code I used to update the auth groups:

The group names had a pattern, so I didn't have to update them one by one.

Run before migrations:
```
postgres=# update auth_group set name = replace(name, 'Long Name Pattern', 'XXX');
```

Run after migrations:

```
postgres=# update auth_group set name = replace(name, 'XXX', 'Long Name Pattern');
```

**Comment by laymonage** [comment]
Thanks for the report, and sorry we didn't get back to you sooner. Migration `0048_add_default_workflows` was added in v2.10 (fc1d857f6339147f063d913704b6cdd36c15dfd5), so this is not something specific to a Wagtail 5 upgrade. It would be useful to know which version you were upgrading from, but given the information it's likely you were upgrading from a version older than 2.10.

We recommend upgrading one feature release at a time, e.g. 2.9 to 2.10, then to 2.11, and so on, to help identify any breakages. At the time this issue was created, 2.10 is already way out of support. So even if we fixed the migration _then_, we wouldn't issue a 2.10.x release, thus we could no longer help the upgrade process.
